### ISSUE TITLE
kotest-spring-extension: Unexpected error from ByteBuddy when using "import" inside package name 

### ISSUE THREAD

**Issue by theHacker** [kotest/kotest#2489]
I hit a strange error message when using Kotest's Spring Extension.
My projects worked like a charm, but when adding a new test, I got this exception.

Simplified reproduction:
```kt
package org.example.myproject.import

import io.kotest.core.spec.style.StringSpec
import org.springframework.boot.test.context.SpringBootTest

@SpringBootTest
class BadPackageTest : StringSpec({

    "empty test should always be green" {
    }
})
```

```stacktrace
Illegal type name: org.example.myproject.import.BadPackageTest$ByteBuddy$Vwc61bWn for class org.example.myproject.import.BadPackageTest$ByteBuddy$Vwc61bWn
java.lang.IllegalStateException: Illegal type name: org.example.myproject.import.BadPackageTest$ByteBuddy$Vwc61bWn for class org.example.myproject.import.BadPackageTest$ByteBuddy$Vwc61bWn
	at net.bytebuddy.dynamic.scaffold.InstrumentedType$Default.validated(InstrumentedType.java:1352)
	at net.bytebuddy.dynamic.scaffold.MethodRegistry$Default.prepare(MethodRegistry.java:519)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.make(SubclassDynamicTypeBuilder.java:213)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.make(SubclassDynamicTypeBuilder.java:204)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3595)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$Delegator.make(DynamicType.java:3819)
	at io.kotest.extensions.spring.SpringTestExtension.method(SpringTestExtension.kt:93)
	at io.kotest.extensions.spring.SpringTestExtension.intercept(SpringTestExtension.kt:54)
	at io.kotest.core.internal.TestCaseExecutor$intercept$execute$1$1.invokeSuspend(TestCaseExecutor.kt:94)
	at io.kotest.core.internal.TestCaseExecutor$intercept$execute$1$1.invoke(TestCaseExecutor.kt)
	at io.kotest.core.internal.TestCaseExecutor$intercept$execute$1$1.invoke(TestCaseExecutor.kt)
	at io.kotest.core.internal.TestCaseExecutor.intercept(TestCaseExecutor.kt:103)
	at io.kotest.core.internal.TestCaseExecutor.execute(TestCaseExecutor.kt:69)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:104)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner.access$runTest(SingleInstanceSpecRunner.kt:34)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$interceptAndRun$2$4.invokeSuspend(SingleInstanceSpecRunner.kt:57)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$interceptAndRun$2$4.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$interceptAndRun$2$4.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.engine.launchers.SequentialTestLauncher$launch$3$1$1.invokeSuspend(SequentialTestLauncher.kt:22)
	(Coroutine boundary)
	at io.kotest.engine.launchers.SequentialTestLauncher.launch(SequentialTestLauncher.kt:20)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner.execute$interceptAndRun(SingleInstanceSpecRunner.kt:55)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$3.invokeSuspend(SingleInstanceSpecRunner.kt:64)
	at io.kotest.engine.spec.SpecExecutor$runTests$run$1.invokeSuspend(SpecExecutor.kt:87)
	at io.kotest.engine.spec.SpecExecutor$interceptSpec$2$1.invokeSuspend(SpecExecutor.kt:105)
	at io.kotest.extensions.spring.SpringTestExtension$intercept$2.invokeSuspend(SpringTestExtension.kt:47)
	at io.kotest.engine.spec.SpecExecutor$interceptSpec$2.invokeSuspend(SpecExecutor.kt:104)
	at io.kotest.engine.spec.SpecExecutor.runTests(SpecExecutor.kt:92)
	at io.kotest.engine.spec.SpecExecutor.runTestsIfAtLeastOneEnabled(SpecExecutor.kt:72)
	at io.kotest.engine.spec.SpecExecutor.execute(SpecExecutor.kt:41)
	at io.kotest.engine.launchers.DefaultSpecLauncher$sequential$3$1$2.invokeSuspend(DefaultSpecLauncher.kt:56)
Caused by: java.lang.IllegalStateException: Illegal type name: org.example.myproject.import.BadPackageTest$ByteBuddy$Vwc61bWn for class org.example.myproject.import.BadPackageTest$ByteBuddy$Vwc61bWn
	at net.bytebuddy.dynamic.scaffold.InstrumentedType$Default.validated(InstrumentedType.java:1352)
	at net.bytebuddy.dynamic.scaffold.MethodRegistry$Default.prepare(MethodRegistry.java:519)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.make(SubclassDynamicTypeBuilder.java:213)
	at net.bytebuddy.dynamic.scaffold.subclass.SubclassDynamicTypeBuilder.make(SubclassDynamicTypeBuilder.java:204)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase.make(DynamicType.java:3595)
	at net.bytebuddy.dynamic.DynamicType$Builder$AbstractBase$Delegator.make(DynamicType.java:3819)
	at io.kotest.extensions.spring.SpringTestExtension.method(SpringTestExtension.kt:93)
	at io.kotest.extensions.spring.SpringTestExtension.intercept(SpringTestExtension.kt:54)
	at io.kotest.core.internal.TestCaseExecutor$intercept$execute$1$1.invokeSuspend(TestCaseExecutor.kt:94)
	at io.kotest.core.internal.TestCaseExecutor$intercept$execute$1$1.invoke(TestCaseExecutor.kt)
	at io.kotest.core.internal.TestCaseExecutor$intercept$execute$1$1.invoke(TestCaseExecutor.kt)
	at io.kotest.core.internal.TestCaseExecutor.intercept(TestCaseExecutor.kt:103)
	at io.kotest.core.internal.TestCaseExecutor.execute(TestCaseExecutor.kt:69)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner.runTest(SingleInstanceSpecRunner.kt:104)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner.access$runTest(SingleInstanceSpecRunner.kt:34)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$interceptAndRun$2$4.invokeSuspend(SingleInstanceSpecRunner.kt:57)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$interceptAndRun$2$4.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.engine.spec.runners.SingleInstanceSpecRunner$execute$interceptAndRun$2$4.invoke(SingleInstanceSpecRunner.kt)
	at io.kotest.engine.launchers.SequentialTestLauncher$launch$3$1$1.invokeSuspend(SequentialTestLauncher.kt:22)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:830)
```

I traced the problem to ByteBuddy's `InstrumentedType.java:1640`, where `isValidIdentifier()` returned `false` for the generated test class. Problem is that my package contains `import`.

It's a Java vs Kotlin issue:
In Java it is forbidden of course, but Kotlin lists `import` only as "soft keyword" (see https://kotlinlang.org/docs/keyword-reference.html#soft-keywords):
> The following tokens act as keywords in the context when they are applicable and can be used as identifiers in other contexts:

For unit tests with Kotest, this had not been a problem.
But when adding `@SpringBootTest`, the test suddenly fails and therefore Kotest is my prime suspect ;)

---

So, not sure, whether you want to improve here. I always hate it, when a library/framework suddenly throws an exception and one has no clue, what's the problem and why it's happening suddenly.

Suggestion:
Perhaps you want to check inside `SpringTestExtension.method()` for an invalid Java identifier, before calling ByteBuddy.
In that case you could throw a more meaningful exception.

---

**Which version of Kotest are you using**
```kt
plugins {
    id("org.springframework.boot") version "2.3.4.RELEASE"
    id("io.spring.dependency-management") version "1.0.10.RELEASE"
    kotlin("jvm") version "1.4.10"
    kotlin("plugin.spring") version "1.4.10"
}

testImplementation("io.kotest:kotest-runner-junit5:4.6.2")
testImplementation("io.kotest.extensions:kotest-extensions-spring:1.0.0")
```

**Comment by sksamuel** [comment]
Since its a generated method, we could replace any soft keywords with softwords + _

**Comment by sksamuel** [comment]
Thanks for the bug report. Turns out we can't stop the error, but we can issue a nicer warning. I've released 1.0.1 with that fix.

**Comment by theHacker** [comment]
Nice. Thank you!
